lib/motors/PID.cpp

```diff
@@ -0,0 +1,56 @@
+#include "PID.hpp"
+
+void PIDController::PIDController_Init()
+{
+  /* Clear controller variables */
+  integrator      = 0.0f;
+  prevError       = 0.0f;
+  differentiator  = 0.0f;
+  prevMeasurement = 0.0f;
+
+  out = 0.0f;
+};
```

Comment:
When this is refactored to a constructor, it should take args for each of the values we want to configure e.g. `kp`, `ki`, `kd`, maybe limits? (though they may be constexprs somewhere). The data members are then initialised in the initialiser list, so the constructor would look something like this:
```c++
PidController::PidController(const core::Float kp, const core::Fload ki, ...) : kp_(kp), ki_(ki), ...
{
/* any other code that needs run goes here, but most likely this is empty */
}
```

Reply:
The values we want to configure:

`kp_, ki_, kd_, tau_, minimum_output_, maximum_output_, minimum_integrator_, maximum_integrator_, sample_time_` 

are all `const core::double` in the hpp file. Should I still initialise them in the constructor in the cpp file? Also where do i initialise `integrator_` and `previous_error_` etc? 

And to check, where should `kp` be `kp_` in the code (same with others)? Is it only the data members in the hpp class that should have the trailing dash?

Reply:
1. yes, you can do that in the initaliser list in the cpp file like `kp_(kp)`
2. `integrator_` should also be initialised in the list, but you can just initialise it to 0 instead of a constructor argument
3. `kp` is used only in the constructor to initialise `kp_`, after that throughout the class you use the class' data member so it is `kp_`

It might also be worth putting all of those into a `PidParameters` struct, just to make the constructor a bit cleaner